--- origsrc/svn/configure.in	2011-08-10 15:16:49.000000000 -0500
+++ src/svn/configure.in	2011-08-10 16:24:22.307142300 -0500
@@ -3,6 +3,7 @@ AC_INIT([ffms2],[2.13.1])
 AC_CONFIG_SRCDIR([src/core/ffms.cpp])
 AC_CONFIG_MACRO_DIR([m4])
 AM_INIT_AUTOMAKE([subdir-objects])
+m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
 if test -z "$CFLAGS"; then
     CFLAGS="-O3"
@@ -114,9 +115,6 @@ CPPFLAGS="$CPPFLAGS -D_FILE_OFFSET_BITS=
 AC_SUBST([LTUNDEF])
 
-SHAVE_INIT([shave], [enable])
 AC_CONFIG_FILES([
-shave/shave
-shave/shave-libtool
 Makefile 
 ffms2.pc
 ])
